src/routes/open/container/Open.tsx

```diff
@@ -82,17 +83,32 @@ export const createSafe = async (values: CreateSafeValues, userAccount: string):
   const ownerAddresses = getAccountsFrom(values)
   const safeCreationSalt = getSafeCreationSaltFrom(values)
   const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
-
-  const receipt = await deploymentTx
+  let receiptResult
+  deploymentTx
     .send({
       from: userAccount,
       gas: values?.gasLimit,
     })
     .once('transactionHash', (txHash) => {
       saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, { txHash, ...values })
+      txMonitor({ sender: userAccount, hash: txHash, data: deploymentTx.encodeABI() }, (txReceipt) => {
+        receiptResult = txReceipt
+      })
+    })
+    .then((txReceipt) => {
+      receiptResult = txReceipt
     })
 
-  return receipt
+  const waitForTxReceipt = (resolve) => {
+    setTimeout(() => {
+      if (receiptResult) {
+        resolve(receiptResult)
+      } else {
+        waitForTxReceipt(resolve)
+      }
+    }, 500)
```

Comment:
The only thing that bothers me about this timeout loop is that it will never exit if the receiptResult doesn't arrive.
Also, does it correctly handle the case when the tx fails?

Reply:
Maybe clear the timeout in the useEffect callback?